modular/gater/admin_handler.go

```diff
@@ -488,6 +487,46 @@ func (g *GateModular) replicateHandler(w http.ResponseWriter, r *http.Request) {
 	log.CtxDebug(reqCtx.Context(), "succeed to replicate piece")
 }
 
+func (g *GateModular) checkReplicatePermission(receiveTask gfsptask.GfSpReceivePieceTask, signatureAddr string) error {
+	ctx := context.Background()
+	// check if the request account is the primary SP of the object of the receiving task
+	bucketInfo, err := g.baseApp.Consensus().QueryBucketInfo(ctx, receiveTask.GetObjectInfo().BucketName)
+	if err != nil {
+		return ErrConsensusWithDetail("QueryBucketInfo error: " + err.Error())
+	}
+
+	gvg, err := g.baseApp.GfSpClient().GetGlobalVirtualGroup(ctx, bucketInfo.Id.Uint64(), receiveTask.GetGlobalVirtualGroupID())
+	if err != nil {
+		return ErrConsensusWithDetail("QueryGVGInfo error: " + err.Error())
+	}
+
+	// judge if sender is the primary sp of the gvg
+	primarySp, err := g.baseApp.Consensus().QuerySPByID(ctx, gvg.PrimarySpId)
+	if err != nil {
+		return ErrConsensusWithDetail("QuerySPInfo error: " + err.Error())
+	}
+
+	if primarySp.GetOperatorAccAddress().String() != signatureAddr {
+		log.CtxErrorw(ctx, "primary sp mismatch", "expect",
+			primarySp.GetOperatorAccAddress().String(), "current", signatureAddr)
+		return ErrPrimaryMismatch
+	}
+
+	// judge if myself is the right secondary sp of the gvg
+	spID, err := g.getSPID()
+	if err != nil {
+		return ErrConsensusWithDetail("getSPID error: " + err.Error())
+	}
+
+	expectSecondarySPID := gvg.GetSecondarySpIds()[int(receiveTask.GetRedundancyIdx())]
+	if expectSecondarySPID != spID {
+		log.CtxErrorw(ctx, "secondary sp mismatch", "expect", expectSecondarySPID, "current", spID)
+		return ErrSecondaryMismatch
+	}
+
```

Comment:
should add the check that the object is creted state, the object state should get from chain or metadata.